### ISSUE TITLE
Some activities not resumable

### ISSUE THREAD

**Issue by Menschheit** [CleverRaven/Cataclysm-DDA#50380]
### Describe the bug

Butchery and digging are not resumable, they should be. (think about it, you want to dig a deep hole and interupted by some zombies, you killed the zombies and must do the work from beginning, then you feel tired, your are slowed ... and you will never get the work done before falling asleep).

The codes assign butchery activity are out dated, they don't work correctly.  Need some fixes or just migrate all the type-dependent activities to new actor class.

The actor class of digging is ok, but the codes that assign the activity forgot to enable the "resumable" flag.

There may be more, but digging is definitely the biggest pain.

### Steps To Reproduce

1. do butchery or digging
2. press 5 to interupt.
3. can't resume the work

### Expected behavior

Many long term activities need to be resumable, otherwise the character can starve to death for digging a hole.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: fc6d52f [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]

**Comment by actual-nh** [comment]
@Saicchi, interested?

**Comment by Saicchi** [comment]
While butchering is not resumable, digging is, so it's probably related to this #47588

**Comment by Menschheit** [comment]
Yes digging is ok. I just need to put the soils into the same tile...
